File: CS-Examples/12_Formulas/XMATCHFunction.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace XMATCHFunction
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load an existing workbook with a pivot table from a file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\XMATCHFunction");

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Set the formula for cell C4 to use XMATCH function
            sheet.Range["C4"].Formula = "=XMATCH(\"Lili\", A2:A5)";

            //Calculate all cells
            workbook.CalculateAllValue();

            // Specify the output file name for the result
            string result = "XMATCHFunction_result.xlsx";

            // Save the modified workbook to the specified file using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }
}

File: VB-Examples/12_Formulas/XMATCHFunction.vb
Imports Spire.Xls
Imports System
Imports System.Windows.Forms

Namespace XMATCHFunction
    Public Partial Class Form1 : Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs)
            ' Create a new workbook
            Dim workbook As Workbook = New Workbook()

            ' Load an existing workbook with a pivot table from a file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\XMATCHFunction")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets[0]

            ' Set the formula for cell C4 to use XMATCH function
            sheet.Range["C4"].Formula = "=XMATCH(\"Lili\", A2:A5)"

            'Calculate all cells
            workbook.CalculateAllValue()

            ' Specify the output file name for the result
            Dim result As String = "XMATCHFunction_result.xlsx"

            ' Save the modified workbook to the specified file using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Dispose of the workbook object to release resources
            workbook.Dispose()

            'View the document
            FileViewer(result)
        End Sub

        Private Sub FileViewer(ByVal fileName As String)
            Try
                System.Diagnostics.Process.Start(fileName)
        Catch
        End Try
        End Sub

        Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs)
            Close()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)

        End Sub
    End Class
End Namespace
